tests/dynamic_data/users_roles/delete_user_test.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+import requests
+from layman import app, settings
+from test_tools import process_client
+from test_tools.util import url_for
+
+
+USERNAME = 'test_delete_user'
+TOKEN_HEADER = 'Authorization'
+
+
+@pytest.mark.usefixtures('ensure_layman_module', 'oauth2_provider_mock')
+def test_delete_user():
+    process_client.reserve_username(USERNAME, actor_name=USERNAME)
+    publication = 'test_delete_user_publication'
+    for publication_type in process_client.PUBLICATION_TYPES:
+        process_client.publish_workspace_publication(publication_type, USERNAME, publication, actor_name=USERNAME)
+
+        # check if publications exists
+        publications = process_client.get_publications(publication_type, workspace=USERNAME, actor_name=USERNAME)
+        assert any(pub.get('name') == publication for pub in publications), f"Publication {publication} was not created"
+
+    # check if workspace exists
+    assert process_client.workspace_exists(USERNAME), f"Workspace '{USERNAME}' was not found"
+    assert process_client.user_exists(USERNAME), f"User '{USERNAME}' was not found"
+
+    with app.app_context():
+        url = url_for('rest_user.delete', username=USERNAME)
+    actor_name = USERNAME
+    headers = {}
+    if actor_name:
+        assert TOKEN_HEADER not in headers
+    if actor_name and actor_name != settings.ANONYM_USER:
+        headers.update(process_client.get_authz_headers(actor_name))
+    response = requests.delete(url, headers=headers, timeout=settings.DEFAULT_CONNECTION_TIMEOUT)
+    assert response.status_code == 200, response.json()
+
+    # check if publication was deleted
+    for publication_type in process_client.PUBLICATION_TYPES:
+        publications_after_delete = process_client.get_publications(publication_type, actor_name=USERNAME)
+        assert not any(pub.get('name') == publication for pub in publications_after_delete), f"Publication {publications_after_delete} was not deleted"
```

Comment:
Please check also that layman.util.get_publication_info returns empty dictionary.